contrib/verifybinaries/verify.py

```diff
@@ -47,136 +189,498 @@ def parse_version_string(version_str):
     return version_base, version_rc, version_os
 
 
-def download_with_wget(remote_file, local_file=None):
-    if local_file:
-        wget_args = ['wget', '-O', local_file, remote_file]
-    else:
-        # use timestamping mechanism if local filename is not explicitly set
-        wget_args = ['wget', '-N', remote_file]
-
-    result = subprocess.run(wget_args,
+def download_with_wget(remote_file, local_file):
+    result = subprocess.run(['wget', '-O', local_file, remote_file],
                             stderr=subprocess.STDOUT, stdout=subprocess.PIPE)
     return result.returncode == 0, result.stdout.decode().rstrip()
 
 
+def download_lines_with_urllib(url) -> (bool, [str]):
+    """Get (success, text lines of a file) over HTTP."""
+    try:
+        return True, [
+            line.strip().decode() for line in urllib.request.urlopen(url).readlines()]
+    except urllib.request.HTTPError as e:
+        log.warning(f"HTTP request to {url} failed (HTTPError): {e}")
+    except Exception as e:
+        log.warning(f"HTTP request to {url} failed ({e})")
+    return (False, [])
+
+
+def verify_with_gpg(
+    signature_filename,
+    output_filename: t.Optional[str] = None
+) -> t.Tuple[int, str]:
+    args = [
+        'gpg', '--yes', '--decrypt',
```

Comment:
```suggestion
        'gpg', '--yes', '--decrypt', '--verify-options', 'show-primary-uid-only',
```

Reply:
This fixes my comment about not strictly checking trusted for keys displaying akas (https://github.com/bitcoin/bitcoin/pull/23020/files#r964163444)